include/swift/AST/DiagnosticsParse.def

```diff
@@ -1316,6 +1316,20 @@ ERROR(attr_availability_expected_equal,none,
 ERROR(attr_availability_expected_version,none,
       "expected version number in '%0' attribute", (StringRef))
 
+WARNING(attr_availability_swift_platform_expected_option,none,
+      "expected '%0' option with a version number for swift platform, "
+      "such as 'introduced', 'deprecated', or 'obsoleted'", (StringRef))
+WARNING(attr_availability_swift_platform_deprecated,none,
+      "'deprecated' must have a version number for swift platform in '%0' "
+      "attribute", (StringRef))
+WARNING(attr_availability_swift_platform_infeasible,none,
+      "argument '%0' is infeasible for swift platform in '%1' attribute",
```

Comment:
Nitpick: To match the tone of other diagnostics, I would prefer something more like "cannot be used with" instead of "is infeasible for". (I'm also not sure we call them "arguments" in attributes, but in this case I think just leading with `'%0'` is fine.)